orion/sites/admin/src/location/location.js

```diff
@@ -20,29 +24,100 @@ export class LocationPage extends React.Component {
         });
     }
 
+    onCheckRow() {
+        const elements = document.getElementsByName("unpublished") || [];
+        let numChecked = 0;
+        for (let i = 0; i < elements.length; i++) {
+            if (elements[i].checked) {
+                numChecked++;
+            }
+        }
+        this.setState({ numSelected: numChecked });
+    }
+
+    onClickSelectAll() {
+        var items = document.getElementsByName("unpublished");
+        for (var i = 0; i < items.length; i++) {
+            if (items[i].type == "checkbox") {
+                items[i].checked = true;
+            }
+        }
+        this.setState({ numSelected: items.length });
+    }
+
+    onClickPublish() {
+        console.log("clicked publish");
+    }
+
     render() {
         const location = this.state.list.map((location, index) => {
-            return <LocationRow key={index} location={location} />;
+            return (
+                <LocationRow
+                    key={index}
+                    location={location}
+                    onCheckRow={this.onCheckRow}
+                />
+            );
         });
         const numLocations = location.length;
         return (
             <div id="view-location">
-                <h1>All Locations ({numLocations})</h1>
+                <div>
+                    <h1>All Locations ({numLocations})</h1>
+                    <p>
+                        You have {this.state.numUnpublished} unpublished items.{" "}
+                        <br />
+                        You have selected {this.state.numSelected} items to
+                        publish.
+                    </p>
+                </div>
                 <ul id="list-heading">
+                    <button
+                        className="li-small"
+                        onClick={this.onClickSelectAll}>
+                        Select All
+                    </button>
                     <li className="li-med">Location ID</li>
                     <li className="li-large">Address</li>
                     <li className="li-large">Room</li>
                 </ul>
                 <ul>{location}</ul>
-                <button id="add-location">
-                    <Link to={"/locations/add"}>Add Location</Link>
-                </button>
+                <div id="list-buttons">
+                    <button>
+                        <Link to={"/locations/add"} id="add-location">
+                            Add Location
+                        </Link>
+                    </button>
+                    <button
+                        id="publish"
+                        className="publish"
```

Comment:
Does this button need to have both `id` and `className`? Probably just className right?